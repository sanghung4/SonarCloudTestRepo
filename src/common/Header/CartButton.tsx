import { useContext } from 'react';

import {
  Badge,
  Box,
  Button,
  IconButton,
  useScreenSize
} from '@dialexa/reece-component-library';
import { CircularProgress } from '@mui/material';
import { useLocation, useHistory } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

import { AuthContext } from 'AuthProvider';

import { cartButtonStyles } from 'common/Header/util';
import { CartIcon } from 'icons';
import { useCartContext } from 'providers/CartProvider';

export default function CartButton() {
  /**
   * Custom hooks
   */
  const location = useLocation();
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();

  /**
   * Context
   */
  const { cartLoading, itemLoading } = useCartContext();

  /**
   * Callbacks
   */
  const goToCart = useGoToCart();

  /**
   * Render
   */
  if (location.pathname !== '/select-accounts') {
    return isSmallScreen ? (
      <Box position="relative" ml={3}>
        {!!(cartLoading || itemLoading) && (
          <cartButtonStyles.CartLoaderComponent>
            <CircularProgress size="2rem" />
          </cartButtonStyles.CartLoaderComponent>
        )}
        <IconButton
          onClick={goToCart}
          size="large"
          data-testid="go-to-cart-button-mobile"
        >
          <CartButtonContent />
        </IconButton>
      </Box>
    ) : (
      <Button
        data-testid="go-to-cart-button-desktop"
        color="gray"
        iconColor="primary"
        startIcon={<CartButtonContent />}
        variant="text"
        onClick={goToCart}
        sx={cartButtonStyles.cartButtonSx}
      >
        {t('common.cart')}
      </Button>
    );
  }
  return null;
}

export function CartButtonContent() {
  /**
   * Hooks / Consts
   */
  const { isSmallScreen } = useScreenSize();
  const { cartLoading, itemCount, itemLoading } = useCartContext();
  const invisible = !!(cartLoading || itemLoading || !itemCount);
  const isLoading = !!(!isSmallScreen && (cartLoading || itemLoading));

  /**
   * Render
   */
  return (
    <Badge
      showZero={false}
      badgeContent={itemCount}
      invisible={invisible}
      max={999}
    >
      <CartIcon height={20} width={20} />
      {isLoading && (
        <cartButtonStyles.CartLoaderComponent>
          <CircularProgress size="2rem" />
        </cartButtonStyles.CartLoaderComponent>
      )}
    </Badge>
  );
}

export function useGoToCart() {
  /**
   * Hooks
   */
  const history = useHistory();
  const { authState } = useContext(AuthContext);
  const { checkingOutWithQuote, clearQuote, contract } = useCartContext();

  /**
   * Output
   */
  return () => {
    if (checkingOutWithQuote) {
      clearQuote();
    }
    const showNavAlert = authState?.isAuthenticated && contract;
    history.push(
      authState?.isAuthenticated ? '/cart' : '/login',
      showNavAlert ? { canShowNavAlert: true } : {}
    );
  };
}
