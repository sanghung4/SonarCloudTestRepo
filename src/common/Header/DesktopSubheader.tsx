import { useContext } from 'react';

import {
  Container,
  Grid,
  Link,
  Toolbar
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';
import { Link as RouterLink } from 'react-router-dom';

import { AuthContext } from 'AuthProvider';
import CategoriesButtonWrapper from 'Categories/CategoriesButtonWrapper';
import SwitchAccounts from 'common/Header/SwitchAccounts';
import {
  desktopSubheaderStyles,
  DesktopSubheaderMenu
} from 'common/Header/util';
import { MaxIcon } from 'icons';
import { useDomainInfo } from 'hooks/useDomainInfo';
import { ErpSystemEnum } from 'generated/graphql';
import { useSelectedAccountsContext } from 'providers/SelectedAccountsProvider';

export default function DesktopSubheader() {
  /**
   * Hooks
   */
  const { t } = useTranslation();
  const { profile } = useContext(AuthContext);
  const { isWaterworks } = useDomainInfo();

  /**
   * Context
   */
  const { selectedAccounts } = useSelectedAccountsContext();
  const isMincron = selectedAccounts.erpSystemName === ErpSystemEnum.Mincron;

  /**
   * Render
   */
  return (
    <Toolbar disableGutters sx={desktopSubheaderStyles.toolbarSx}>
      <Container maxWidth="lg">
        <Grid container alignItems="center">
          {/* ===== Browse Products ===== */}
          {!isWaterworks && !isMincron && (
            <Grid item>
              <CategoriesButtonWrapper />
            </Grid>
          )}
          {/* ===== Brands ===== */}
          {!isWaterworks && !isMincron && (
            <desktopSubheaderStyles.GridStyledLinks item>
              <Link
                component={RouterLink}
                underline="none"
                to="/brands"
                data-testid="nav-brands"
              >
                {t('common.brands')}
              </Link>
            </desktopSubheaderStyles.GridStyledLinks>
          )}
          {/* ===== MAX ===== */}
          <DesktopSubheaderMenu item="max">
            <MaxIcon />
          </DesktopSubheaderMenu>
          {/* ===== Resources ===== */}
          <DesktopSubheaderMenu item="resources">
            {t('common.resources')}
          </DesktopSubheaderMenu>
          {!!profile?.userId && (
            <Grid item ml="auto" mr={-1}>
              <SwitchAccounts />
            </Grid>
          )}
        </Grid>
      </Container>
    </Toolbar>
  );
}
