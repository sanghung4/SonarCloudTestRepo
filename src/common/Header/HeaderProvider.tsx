import { ReactNode, createContext, useState } from 'react';

type Props = {
  children: ReactNode;
};

type HeaderContextType = {
  searchOpen: boolean;
  setSearchOpen: (value: boolean) => void;
  searchPage: number;
  setSearchPage: (value: number) => void;
  trackedSearchTerm: string;
  setTrackedSearchTerm: (value: string) => void;
  pageIndex: number;
  setPageIndex: (value: number) => void;
};

export const HeaderContext = createContext<HeaderContextType>({
  searchOpen: false,
  setSearchOpen: () => {},
  searchPage: 0,
  setSearchPage: () => {},
  trackedSearchTerm: '',
  setTrackedSearchTerm: () => {},
  pageIndex: 0,
  setPageIndex: () => {}
});

export default function HeaderProvider(props: Props) {
  /**
   * State
   */
  const [searchOpen, setSearchOpen] = useState(false);
  const [searchPage, setSearchPage] = useState(0);
  const [trackedSearchTerm, setTrackedSearchTerm] = useState('');
  const [pageIndex, setPageIndex] = useState(0);

  /**
   * Provider Render
   */
  return (
    <HeaderContext.Provider
      value={{
        searchOpen,
        setSearchOpen,
        searchPage,
        setSearchPage,
        trackedSearchTerm,
        setTrackedSearchTerm,
        pageIndex,
        setPageIndex
      }}
    >
      {props.children}
    </HeaderContext.Provider>
  );
}
