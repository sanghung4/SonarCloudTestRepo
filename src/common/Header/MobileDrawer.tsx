import { useContext, useState } from 'react';

import {
  Box,
  Button,
  Grid,
  IconButton,
  List,
  ListItem,
  SwipeableDrawer,
  Typography,
  Divider
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';

import { AuthContext } from 'AuthProvider';
import CategoriesMobileWrapper from 'Categories/CategoriesMobileWrapper';
import SwitchAccounts from 'common/Header/SwitchAccounts';
import {
  handleCategoriesClose,
  MobileDrawerCollapseCompanyList,
  MobileDrawerCollapseList,
  mobileDrawerStyles,
  mobileOption
} from 'common/Header/util';
import HoldAlert from 'common/HoldAlert';
import { CloseIcon, MaxIcon, SignOutIcon } from 'icons';
import { useDomainInfo } from 'hooks/useDomainInfo';
import useUserAgent from 'hooks/useUserAgent';
import { useSelectedAccountsContext } from 'providers/SelectedAccountsProvider';

type Props = {
  open: boolean;
  setOpen: (val: boolean) => void;
};

export default function MobileDrawer(props: Props) {
  /**
   * Custom Hooks
   */
  const history = useHistory();
  const { t } = useTranslation();
  const { isiOS } = useUserAgent();
  const { isWaterworks } = useDomainInfo();

  /**
   * Context
   */
  const { activeFeatures, authState, firstName, profile } =
    useContext(AuthContext);
  const { selectedAccounts, error: erpAccountError } =
    useSelectedAccountsContext();

  /**
   * State
   */
  const [profileOpen, setProfileOpen] = useState(false);
  const [maxOpen, setMaxOpen] = useState(false);
  const [resourcesOpen, setResourcesOpen] = useState(false);
  const [showCategories, setShowCategories] = useState(false);

  /**
   * Styles
   */
  const { drawerPaperSx, listSx } = mobileDrawerStyles;

  /**
   * Render
   */
  return (
    <SwipeableDrawer
      PaperProps={{ sx: drawerPaperSx }}
      open={props.open}
      disableBackdropTransition={!isiOS}
      disableDiscovery={isiOS}
      onClose={handleSetOpen(false)}
      onOpen={handleSetOpen(true)}
      data-testid="mobile-menu-drawer"
    >
      <CategoriesMobileWrapper
        in={showCategories}
        handleClose={handleCategoriesClose(setShowCategories, props.setOpen)}
      />
      <Grid
        container
        wrap="nowrap"
        pt={1.5}
        pb={3}
        display="flex"
        flexDirection="column"
        minHeight={1}
      >
        <Box width={1}>
          {/* ===== Title Bar ===== */}
          <Grid container item justifyContent="flex-end" flexGrow={1} px={3}>
            <IconButton
              onClick={handleSetOpen(false)}
              size="large"
              sx={{ p: 0 }}
              data-testid="mobile-close-button"
            >
              <CloseIcon />
            </IconButton>
          </Grid>
          {/* ===== Accounts ===== */}
          {!!profile?.userId && (
            <>
              <Typography variant="h5" fontWeight={500} px={3} pb={2}>
                {t('common.hello', { name: firstName })}!
              </Typography>
              <Box component={Divider} px={3} />
              {!!(
                selectedAccounts.billToErpAccount?.creditHold ||
                selectedAccounts.shipToErpAccount?.creditHold
              ) && (
                <Box pt={1} px={3}>
                  <HoldAlert />
                </Box>
              )}
              <Box px={3}>
                <SwitchAccounts onChange={handleSetOpen(false)} />
              </Box>
            </>
          )}
        </Box>
        <Box flex={1} overflow="auto" mt={authState?.isAuthenticated ? 0 : 1}>
          <List sx={listSx}>
            {/* ********** Browse Products ********** */}
            {!isWaterworks && (
              <ListItem
                button
                onClick={handleSetShowCategories(true)}
                data-testid="browse-products-button"
              >
                {t('common.browseProducts')}
              </ListItem>
            )}
            {/* ********** Lists ********** */}
            {!!activeFeatures?.includes('LISTS') && !isWaterworks && (
              <ListItem
                button
                onClick={mobileOptionHandler('/lists', true)}
                data-testid="my-lists-button"
              >
                {t('common.myLists')}
              </ListItem>
            )}
            {/* ********** MAX ********** */}
            <MobileDrawerCollapseList
              item="max"
              open={maxOpen}
              setOpen={setMaxOpen}
              urlHandler={mobileOptionHandler}
            >
              <MaxIcon />
            </MobileDrawerCollapseList>
            {/* ********** My Profile ********** */}
            {!authState?.isAuthenticated ? (
              <ListItem
                button
                onClick={mobileOptionHandler('/login')}
                data-testid="sign-in-header-button"
              >
                {t('common.signInRegister')}
              </ListItem>
            ) : (
              !erpAccountError && (
                <MobileDrawerCollapseList
                  item="profile"
                  open={profileOpen}
                  setOpen={setProfileOpen}
                  urlHandler={mobileOptionHandler}
                >
                  {t('common.myProfile')}
                </MobileDrawerCollapseList>
              )
            )}
            {/* ********** Resources ********** */}
            <MobileDrawerCollapseList
              item="resources"
              open={resourcesOpen}
              setOpen={setResourcesOpen}
              urlHandler={mobileOptionHandler}
            >
              {t('common.resources')}
            </MobileDrawerCollapseList>
            {/* ********** Brands ********** */}
            {!isWaterworks && (
              <ListItem
                button
                onClick={mobileOptionHandler('/brands')}
                data-testid="brands-button"
              >
                {t('common.brands')}
              </ListItem>
            )}
            {/* ********** Store Finder ********** */}
            <ListItem
              button
              onClick={mobileOptionHandler('/location-search')}
              data-testid="store-finder-button"
            >
              {t('common.storeFinder')}
            </ListItem>
            {/* ********** Support ********** */}
            <ListItem
              button
              onClick={mobileOptionHandler('/support')}
              data-testid="help-button"
            >
              {t('support.help')}
            </ListItem>
            {/* ********** Companies ********** */}
            <MobileDrawerCollapseCompanyList />
          </List>
        </Box>
        {/* ********** LOG OUT ********** */}
        {!!authState?.isAuthenticated && (
          <Box pt={2} display="flex" justifyContent="flex-end" width={1} pr={1}>
            <Button
              startIcon={<SignOutIcon />}
              onClick={mobileOptionHandler('/logout')}
              variant="text"
            >
              <Typography
                variant="body1"
                pl={1.5}
                color="primary02.main"
                data-testid="sign-out-button"
              >
                {t('common.signOut')}
              </Typography>
            </Button>
          </Box>
        )}
      </Grid>
    </SwipeableDrawer>
  );

  /**
   * Handles
   */
  function mobileOptionHandler(path: string, auth?: boolean) {
    return () =>
      mobileOption(
        resetOpen,
        history.push,
        authState?.isAuthenticated,
        path,
        auth
      );
  }
  function handleSetOpen(show: boolean) {
    return () => props.setOpen(show);
  }
  function handleSetShowCategories(show: boolean) {
    return () => setShowCategories(show);
  }

  /**
   * Misc
   */
  function resetOpen() {
    props.setOpen(false);
    setProfileOpen(false);
    setMaxOpen(false);
    setResourcesOpen(false);
    setProfileOpen(false);
  }
}
