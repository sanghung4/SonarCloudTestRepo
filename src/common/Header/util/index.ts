export { DesktopSubheaderMenu } from './DesktopSubheaderMenu';
export {
  handleCategoriesClose,
  mobileOption,
  useResizeObserverRoundFn
} from './etc';
export {
  MobileDrawerCollapseList,
  MobileDrawerCollapseCompanyList
} from './MobileDrawerCollapseList';
export { getNavData } from './navData';
export {
  cartButtonStyles,
  CompanyListButton,
  desktopSubheaderStyles,
  indexStyles,
  mobileDrawerStyles,
  ReeceLogo,
  switchAccountsStyles,
  userButtonStyles
} from './styles';
export { UserItemButton } from './UserItemButton';

export type { NavData, NavItem } from './navData';
