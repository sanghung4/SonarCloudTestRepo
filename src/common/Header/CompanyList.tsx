import { useContext } from 'react';
import {
  Box,
  Container,
  DropdownButton,
  Grid,
  Hidden,
  Link,
  MenuItem
} from '@dialexa/reece-component-library';
import { useHistory, useLocation } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

import UserButton from 'common/Header/UserButton';
import { CompanyListButton } from 'common/Header/util';
import { HelpIcon, LocationIcon } from 'icons';
import { Configuration } from 'utils/configuration';
import { useDomainInfo } from 'hooks/useDomainInfo';
import { generateCompanyUrl } from 'utils/brandList';
import { AuthContext } from 'AuthProvider';
import {
  CompanyDomain,
  expressionsHomeGallery,
  isWaterworksEnabled,
  isExpressionsHomeGalleryEnabled,
  waterworksCompany
} from 'hooks/utils/useDomainInfo';

function CompanyListHeader() {
  /**
   * Custom Hooks
   */
  const { companyList, port } = useDomainInfo();
  const history = useHistory();
  const location = useLocation();
  const { t } = useTranslation();

  /**
   * Context
   */
  const { features } = useContext(AuthContext);

  /**
   * Logic
   */
  let filteredCompanyList: CompanyDomain[] = [];
  if (!isWaterworksEnabled(features)) {
    filteredCompanyList = companyList.filter(
      (c) => c.sub !== waterworksCompany.sub
    );
  } else {
    filteredCompanyList = companyList;
  }

  if (!isExpressionsHomeGalleryEnabled(features)) {
    filteredCompanyList = filteredCompanyList.filter(
      (c) => c.sub !== expressionsHomeGallery.sub
    );
  }

  // Hide if on the select accounts page
  if (
    location.pathname.includes('/select-accounts') ||
    location.pathname.includes('/register')
  ) {
    return null;
  }

  /**
   * Render
   */
  return (
    <Hidden mdDown>
      <Box
        displayPrint="none"
        bgcolor="common.white"
        borderBottom={1}
        borderColor="lighterGray.main"
      >
        <Box component={Container} maxWidth="lg" px={2}>
          <Grid
            py={1}
            container
            justifyContent="space-between"
            alignItems="center"
          >
            <Grid item>
              <DropdownButton
                anchorOrigin={{
                  vertical: 'bottom',
                  horizontal: 'left'
                }}
                data-testid="company-list"
                transformOrigin={{
                  vertical: 'top',
                  horizontal: 'left'
                }}
                content={() => (
                  <Box py={2}>
                    {filteredCompanyList.map((company, i) => (
                      <Link
                        key={i}
                        data-testid={`${company.sub}-link`}
                        underline="none"
                        href={generateCompanyUrl(
                          company.sub,
                          port,
                          Configuration.environment,
                          isWaterworksEnabled(features) ?? false
                        )}
                      >
                        <MenuItem>{company.link}</MenuItem>
                      </Link>
                    ))}
                  </Box>
                )}
              >
                {t('common.reeceCompanies')}
              </DropdownButton>
            </Grid>
            <Grid item>
              <UserButton />
              <CompanyListButton
                color="gray"
                iconColor="primary"
                startIcon={<LocationIcon height={12} width={12} />}
                variant="text"
                onClick={() => {
                  history.push('/location-search');
                }}
                data-testid="location-search-button"
              >
                {t('common.storeFinder')}
              </CompanyListButton>
              <CompanyListButton
                color="gray"
                iconColor="primary"
                startIcon={<HelpIcon />}
                variant="text"
                onClick={() => {
                  history.push('/support');
                }}
                data-testid="support-button"
              >
                {t('support.help')}
              </CompanyListButton>
            </Grid>
          </Grid>
        </Box>
      </Box>
    </Hidden>
  );
}

export default CompanyListHeader;
