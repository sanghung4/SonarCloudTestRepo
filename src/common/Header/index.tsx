import { useContext, useEffect, useState } from 'react';

import {
  AppBar,
  Box,
  Button,
  Container,
  Divider,
  Grid,
  Hidden,
  IconButton,
  Link,
  Toolbar,
  useScreenSize
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';
import { Link as RouterLink, useHistory, useLocation } from 'react-router-dom';

import { AuthContext } from 'AuthProvider';
import BranchButton from 'Branches/BranchButton';
import CartButton from 'common/Header/CartButton';
import DesktopSubheader from 'common/Header/DesktopSubheader';
import MobileDrawer from 'common/Header/MobileDrawer';
import { indexStyles, ReeceLogo } from 'common/Header/util';
import { MenuIcon, SearchIcon, StarFilledIcon } from 'icons';
import logoSrc from 'images/logo.svg';
import { useDomainInfo } from 'hooks/useDomainInfo';
import SearchBar from './SearchBar';
import { HeaderContext } from './HeaderProvider';
import { useSelectedAccountsContext } from 'providers/SelectedAccountsProvider';
import { SelectAccountsHeaderContainer } from './util/styles';

function Header() {
  /**
   * Custom Hooks
   */
  const { isWaterworks, subdomain } = useDomainInfo();
  const history = useHistory();
  const location = useLocation();
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();

  /**
   * State
   */
  const [drawerOpen, setDrawerOpen] = useState(false);

  /**
   * Context
   */
  const { authState, profile } = useContext(AuthContext);
  const { searchOpen, setSearchOpen } = useContext(HeaderContext);
  const { isMincron } = useSelectedAccountsContext();

  /**
   * Callback
   */

  /**
   * Effect
   */
  useEffect(() => {
    document.body.classList.toggle('overflow-hidden', searchOpen);
  }, [searchOpen]);

  /**
   * Styles
   */
  const { containerSx, toolbarSx } = indexStyles(searchOpen, isSmallScreen);

  if (
    location.pathname.includes('/select-accounts') ||
    location.pathname.includes('/register')
  ) {
    return (
      <Box displayPrint="none">
        <Box zIndex="drawer" bottom={0} bgcolor="transparent">
          <AppBar color="inherit" position="static" elevation={0}>
            <Toolbar disableGutters sx={toolbarSx}>
              <Container maxWidth="lg" sx={containerSx}>
                <SelectAccountsHeaderContainer>
                  <ReeceLogo alt="Reece" src={logoSrc} />
                  {subdomain !== 'reece' && (
                    <>
                      <Box mx={1.5} my={isSmallScreen ? 0 : 1}>
                        <Divider orientation="vertical" />
                      </Box>
                      <ReeceLogo
                        alt={subdomain}
                        src={`/logos/${subdomain}.svg`}
                      />
                    </>
                  )}
                </SelectAccountsHeaderContainer>
              </Container>
            </Toolbar>
          </AppBar>
        </Box>
      </Box>
    );
  }

  /**
   * Render
   */
  return (
    <Box displayPrint="none">
      <Hidden mdUp>
        <Box height={searchOpen ? 0 : 'auto'}></Box>
      </Hidden>
      <MobileDrawer open={drawerOpen} setOpen={setDrawerOpen} />
      <Box zIndex="drawer"  bottom={0} bgcolor="transparent" sx = {
        {...(isSmallScreen && searchOpen && { position:"absolute", top:0, width:"100%"})}
      }>
        <AppBar color="inherit" position="static" elevation={0}>
          <Box bgcolor="common.white">
            <Toolbar disableGutters sx={toolbarSx}>
              {!isWaterworks && !isMincron && (
                <Hidden mdUp>
                  <SearchBar />
                </Hidden>
              )}
              <Container maxWidth="lg" sx={containerSx}>
                <Grid container spacing={isSmallScreen ? 0 : 2} wrap="nowrap">
                  <Grid
                    container
                    item
                    md={isSmallScreen ? 4 : 8}
                    justifyContent={isSmallScreen ? 'space-between' : undefined}
                    alignItems="center"
                    wrap="nowrap"
                    position="relative"
                  >
                    <Hidden mdUp>
                      <Box display="flex" flex="1" justifyContent="center">
                        <IconButton
                          onClick={() => setDrawerOpen(true)}
                          size="large"
                          sx={{ mr: 'auto' }}
                          data-testid="mobile-hamburger-menu"
                        >
                          <MenuIcon />
                        </IconButton>
                      </Box>
                    </Hidden>
                    <Box
                      lineHeight="1"
                      display="flex"
                      flex={isSmallScreen ? '0 1 auto' : '0 1 25%'}
                    >
                      <Link
                        to="/"
                        component={RouterLink}
                        sx={{ display: 'flex' }}
                      >
                        <ReeceLogo alt="Reece" src={logoSrc} />
                      </Link>

                      {subdomain !== 'reece' && (
                        <>
                          <Box mx={1.5} my={isSmallScreen ? 0 : 1}>
                            <Divider orientation="vertical" />
                          </Box>
                          <Link
                            to="/"
                            component={RouterLink}
                            sx={{ display: 'flex' }}
                          >
                            <ReeceLogo
                              alt={subdomain}
                              src={`/logos/${subdomain}.svg`}
                            />
                          </Link>
                        </>
                      )}
                    </Box>
                    {!isWaterworks && !isMincron && (
                      <Hidden mdDown>
                        <Box ml={5} display="flex" flex={1}>
                          <SearchBar />
                        </Box>
                      </Hidden>
                    )}
                    <Hidden mdUp>
                      <Box display="flex" flex="1" justifyContent="flex-end">
                        {!isWaterworks && !isMincron && (
                          <IconButton
                            onClick={() => setSearchOpen(true)}
                            size="large"
                            sx={{ ml: 'auto' }}
                            data-testid="header-search-button"
                          >
                            <SearchIcon />
                          </IconButton>
                        )}
                        <CartButton />
                      </Box>
                    </Hidden>
                  </Grid>
                  <Hidden mdDown>
                    <Grid
                      container
                      item
                      md={4}
                      justifyContent="flex-end"
                      alignItems="center"
                      wrap="nowrap"
                    >
                      {location.pathname !== '/select-accounts' && (
                        <>
                          {!isWaterworks && !isMincron && (
                            <Button
                              variant="text"
                              color="gray"
                              iconColor="primary"
                              startIcon={<StarFilledIcon />}
                              onClick={handleListsClick}
                              data-testid="my-lists-button"
                              sx={{
                                fontSize: '14px',
                                lineHeight: '18px',
                                fontWeight: 400,
                                whiteSpace: 'nowrap'
                              }}
                            >
                              {t('common.myLists')}
                            </Button>
                          )}
                          <BranchButton data-testid="branch-button-desktop" />{' '}
                          <CartButton />
                        </>
                      )}
                    </Grid>
                  </Hidden>
                </Grid>
              </Container>
            </Toolbar>
            {!!(
              authState?.isAuthenticated &&
              (!profile?.isEmployee ||
                (profile?.isEmployee && profile?.isVerified)) &&
              location.pathname !== '/select-accounts'
            ) && (
              <Hidden mdUp>
                <Divider />
                <Toolbar disableGutters>
                  <BranchButton data-testid="branch-button-mobile" />
                </Toolbar>
              </Hidden>
            )}
          </Box>
          <Hidden mdDown>
            <DesktopSubheader />
          </Hidden>
        </AppBar>
      </Box>
  </Box>
  );

  function handleListsClick() {
    history.push(authState?.isAuthenticated ? '/lists' : '/login');
  }
}

export default Header;
