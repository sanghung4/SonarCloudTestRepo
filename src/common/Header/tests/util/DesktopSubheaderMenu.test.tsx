import { act, fireEvent } from '@testing-library/react';
import {
  authConfigLoggedIn,
  authConfigLoggedOut
} from 'common/Header/tests/mocks';
import { DesktopSubheaderMenu } from 'common/Header/util';
import { ErpSystemEnum } from 'generated/graphql';
import { render } from 'test-utils/TestWrapper';
import { SelectedAccounts } from 'providers/SelectedAccountsProvider';

const mockChild = 'test';
const mockAccount: SelectedAccounts = {
  billTo: { erpAccountId: '' },
  shipTo: { erpAccountId: '' },
  shippingBranchId: '',
  erpSystemName: ErpSystemEnum.Eclipse
};

describe('common - Header - util - DesktopSubheaderMenu', () => {
  it('should match snapshot for max under Eclipse as logged out user', async () => {
    const { container } = render(
      <DesktopSubheaderMenu item="max">{mockChild}</DesktopSubheaderMenu>,
      {
        authConfig: authConfigLoggedOut,
        selectedAccountsConfig: { selectedAccounts: mockAccount }
      }
    );
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot for max under Eclipse as logged in user', async () => {
    const { container } = render(
      <DesktopSubheaderMenu item="max">{mockChild}</DesktopSubheaderMenu>,
      {
        authConfig: authConfigLoggedIn,
        selectedAccountsConfig: { selectedAccounts: mockAccount }
      }
    );
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot for max under Mincron as logged in user', async () => {
    const { container } = render(
      <DesktopSubheaderMenu item="max">{mockChild}</DesktopSubheaderMenu>,
      {
        authConfig: authConfigLoggedIn,
        selectedAccountsConfig: {
          selectedAccounts: {
            ...mockAccount,
            erpSystemName: ErpSystemEnum.Mincron
          }
        }
      }
    );
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot for resources to click on an item', async () => {
    const { container, getByTestId, getAllByText } = render(
      <DesktopSubheaderMenu item="resources">{mockChild}</DesktopSubheaderMenu>,
      {
        authConfig: authConfigLoggedIn,
        selectedAccountsConfig: { selectedAccounts: mockAccount }
      }
    );
    await act(() => new Promise((res) => setTimeout(res, 0)));
    fireEvent.click(getAllByText(mockChild)[0]);
    await act(() => new Promise((res) => setTimeout(res, 100)));
    fireEvent.click(getByTestId('desktop-nav-credit-forms'));
    await act(() => new Promise((res) => setTimeout(res, 0)));
    expect(container).toMatchSnapshot();
  });
});
