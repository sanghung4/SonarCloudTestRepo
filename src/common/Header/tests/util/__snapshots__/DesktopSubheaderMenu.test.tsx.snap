// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common - Header - util - DesktopSubheaderMenu should match snapshot for max under Eclipse as logged in user 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item css-ajt0mp-MuiGrid-root"
  >
    <button
      aria-haspopup="true"
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-gfygn0-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      test
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium css-1f03u4k-MuiButton-endIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`common - Header - util - DesktopSubheaderMenu should match snapshot for max under Eclipse as logged out user 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item css-ajt0mp-MuiGrid-root"
  >
    <button
      aria-haspopup="true"
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-gfygn0-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      test
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium css-1f03u4k-MuiButton-endIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`common - Header - util - DesktopSubheaderMenu should match snapshot for max under Mincron as logged in user 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item css-ajt0mp-MuiGrid-root"
  >
    <button
      aria-haspopup="true"
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-gfygn0-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      test
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium css-1f03u4k-MuiButton-endIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`common - Header - util - DesktopSubheaderMenu should match snapshot for resources to click on an item 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item css-ajt0mp-MuiGrid-root"
  >
    <button
      aria-haspopup="true"
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-gfygn0-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      test
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium css-1f03u4k-MuiButton-endIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
